From 476c096d858f414e6d3b854c85cf98defc14c664 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Wed, 31 Aug 2022 16:09:17 +0200
Subject: [PATCH 0180/1141] arm64: dts: st: add NVMEM provider to soc
 information

Add NVMEM provider to access on SoC information saved on BSEC OTP:
- RPN = Device part number (BSEC_OTP_DATA9)
- PKG = package data register (Bits 2:0 of BSEC_OTP_DATA122)

Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Ie757042c64e62408dfdcd598d44b5bf710deb12b
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/264788
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/302171
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -787,6 +787,15 @@
 			reg = <0x44000000 0x1000>;
 			#address-cells = <1>;
 			#size-cells = <1>;
+
+			part_number_otp@24 {
+				reg = <0x24 0x4>;
+			};
+
+			package_otp@1e8 {
+				reg = <0x1e8 0x1>;
+				bits = <0 3>;
+			};
 		};
 
 		rcc: rcc@44200000 {
